api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java

```diff
@@ -241,6 +249,36 @@ void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForDwpHouseholdIde
         wiremockManager.assertThatEligibilityRequestMade();
     }
 
+    @Test
+    void shouldReturnDuplicateStatusWhenTwoEligibleClaimsForSameNino() throws JsonProcessingException, InterruptedException {
+        //Given
+        NewClaimDTO dto = aValidClaimDTO();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse
+                = anIdMatchedEligibilityConfirmedUCResponseWithAllMatchesAndDwpHouseIdentifier(DWP_HOUSEHOLD_IDENTIFIER);
+        wiremockManager.stubEligibilityResponse(identityAndEligibilityResponse);
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        //When
+        List<Future<Object>> futures = executor.invokeAll(List.of(
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class),
+                () -> restTemplate.exchange(buildCreateClaimRequestEntity(dto), ClaimResultDTO.class)));
+
+        List<ResponseEntity<ClaimResultDTO>> results = futures.stream().map(objectFuture -> {
+            try {
+                return (ResponseEntity<ClaimResultDTO>) objectFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toList());
```

Comment:
Using `CompletableFutures` here would make this a lot tidier, can I suggest trying to use those instead?